asab/web/authn/middleware.py

```diff
@@ -61,10 +68,17 @@ def authn_optional_handler(func):
 
 	async def wrapper(*args, **kargs):
 		request = args[-1]
+
 		try:
 			kargs['identity'] = request.Identity
 		except AttributeError:
 			kargs['identity'] = None
+
+		try:
+			kargs['oauth_user_info'] = request.OAuthUserInfo
+		except AttributeError:
+			kargs['oauth_user_info'] = None
+
```

Comment:
No, this is not needed ... and it will add an additional requirement to for a signature of the method.
The value is available in `request.OAuthUserInfo`.